src/modules/sub-items/components/table-view/table.view.columns.toggler.list.element.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const TableViewColumnsTogglerListElement = ({ label, isColumnVisible, toggleColumnVisibility, columnKey }) => {
+    return (
+        <li className="c-table-view-columns-toggler-list-element" onClick={() => toggleColumnVisibility(columnKey)}>
```

Comment:
I've been thinking about making it a class component so `() => toggleColumnVisibility(columnKey)` could be invoked inside a class method. That will prevent from additional re-rendering a whole component just because of creating a new function. Keep in mind that final effect of an arrow function as value of any element attribute is similar to function binding, which is a creation of a new function reference.
Sumarizing, I suggest convert this component to a class component and create a method named: `onClickHandler` or similarly:
```
onClickHandler() {
    this.props.toggleColumnVisibility(this.props.columnKey);
}
```

Does it make sense?

Reply:
I made it functional component because we always make them functional if they don't have state and AFAIR we quite often create functions inside functional component and then pass them to props. Another thing is, that a few months ago I suggested the same approach as you are suggesting now and it was rejected.